generators/webapp/templates/server/plugins/webapp/index.js

```diff
@@ -21,33 +21,59 @@ function loadAssetsFromStats(statsFilePath) {
     .then(require)
     .then((stats) => {
       const assets = {};
-      _.each(stats.assetsByChunkName.main, (v) => {
-        if (v.endsWith(".js")) {
-          assets.js = v;
-        } else if (v.endsWith(".css")) {
-          assets.css = v;
+      _.each(stats.assets, (asset) => {
+        const name = asset.name;
+        if (name.startsWith("bundle")) {
+          assets.js = name;
+        } else if (name.endsWith(".css")) {
+          assets.css = name;
+        } else if (name.endsWith("manifest.json")) {
+          assets.manifest = name;
         }
       });
       return assets;
     })
     .catch(() => ({}));
 }
 
+function getIconStats(iconStatsPath) {
+  let iconStats;
+  try {
+    iconStats = fs.readFileSync(Path.resolve(iconStatsPath)).toString();
+    iconStats = JSON.parse(iconStats);
+  } catch (err) {
+    // noop
+  }
+  if (iconStats && iconStats.html) {
+    return iconStats.html.join("");
+  }
+  return iconStats;
+}
+
+/* eslint max-statements: [2, 16] */
 function makeRouteHandler(options, userContent) {
   const CONTENT_MARKER = "{{SSR_CONTENT}}";
-  const BUNDLE_MARKER = "{{WEBAPP_BUNDLES}}";
+  const HEADER_BUNDLE_MARKER = "{{WEBAPP_HEADER_BUNDLES}}";
+  const BODY_BUNDLE_MARKER = "{{WEBAPP_BODY_BUNDLES}}";
   const TITLE_MARKER = "{{PAGE_TITLE}}";
   const PREFETCH_MARKER = "{{PREFETCH_BUNDLES}}";
+  const META_TAGS_MARKER = "{{META_TAGS}}";
   const WEBPACK_DEV = options.webpackDev;
   const RENDER_JS = options.renderJS;
   const RENDER_SS = options.serverSideRendering;
   const html = fs.readFileSync(Path.join(__dirname, "index.html")).toString();
   const assets = options.__internals.assets;
   const devJSBundle = options.__internals.devJSBundle;
   const devCSSBundle = options.__internals.devCSSBundle;
+  const iconStats = getIconStats(options.iconStats);
 
   /* Create a route handler */
+  /*eslint max-statements: 0*/
   return (request, reply) => {
+    if (global.navigator) {
```

Comment:
its required by material-ui if we render server side so it has a better understanding of the screen size. its an optional param. ill take it off since material-ui is not needed for every project.